app/src/androidTest/java/app/cybrid/demoapp/ui/tradeFlow/TradeFlowActivityTest.kt

```diff
@@ -40,13 +37,17 @@ class TradeFlowActivityTest {
     }
 
     @Test
-    fun checkListComponent() {
+    fun checkAppFlow() = runTest {
+
+        Thread.sleep(2500)
+        onView(withId(R.id.demo)).check(matches(isDisplayed())).perform(click())
 
         Thread.sleep(5000)
         onView(withId(R.id.list)).check(matches(isDisplayed()))
         onView(withText("TradeFlow")).check(matches(isDisplayed())).perform(click())
 
         Thread.sleep(5000)
+        compose.waitForIdle()
```

Comment:
Sleeps tend to be problematic. Does the test framework allow you to wait on components to load?

Reply:
Waits for compose to be idle.

This is a blocking call. Returns only after compose is idle.
In case the main clock auto advancement is enabled (by default is) this will also keep advancing the clock until it is idle (meaning there are no recompositions, animations, etc. pending). If not, this will wait only for other idling resources.

At the beginning of the test file i create a composeEmptyRule and then when the activity triggers to the autoclock that the UI is handled by compose, the compose Rule become active!!